src/m365/teams/commands/channel/channel-member-add.ts

```diff
@@ -200,7 +200,11 @@ class TeamsChannelMemberAddCommand extends GraphCommand {
           return Promise.reject(`The specified channel '${args.options.channelName}' does not exist in the Microsoft Teams team with ID '${teamId}'`);
         }
 
-        return Promise.resolve(channelItem.id);
+        if (channelItem.membershipType !== "private") {
+          return Promise.reject(`The specified channel is not a private channel`);
+        }
+
+        return Promise.resolve(channelItem.id!);
       }, err => { return Promise.reject(err); });
```

Comment:
We don't need to handle erroneous promise here if all we do is reject it. That's the default behavior that needs to be handled by the caller, so we can skip this part.

Reply:
nice catch!